crates/sargon/src/profile/mfa/security_structures/security_shield_builder.rs

```diff
@@ -53,6 +53,81 @@ impl Default for SecurityShieldBuilder {
     }
 }
 
+impl PartialEq for SecurityShieldBuilder {
+    fn eq(&self, other: &Self) -> bool {
+        let (matrix, name, authentication_signing_factor) = (
+            self.matrix_builder
+                .read()
+                .expect("Failed to read matrix_builder"),
+            self.name.read().expect("Failed to read name"),
```

Comment:
we might wanna use `try_read` here? but we can merge without. Neither Eq nor Clone impl should blocking wait